alert.go

```diff
@@ -910,6 +916,31 @@ func (a *AlertNode) handlePost(post *pipeline.PostHandler, ad *AlertData) {
 	return
 }
 
+func (a *AlertNode) handleTcp(tcp *pipeline.TcpHandler, ad *AlertData) {
+	bodyBuffer := a.bufPool.Get().(*bytes.Buffer)
+	defer func() {
+		bodyBuffer.Reset()
+		a.bufPool.Put(bodyBuffer)
+	}()
+
+	err := json.NewEncoder(bodyBuffer).Encode(ad)
+	if err != nil {
+		a.logger.Println("E! failed to marshal alert data json", err)
+		return
+	}
+
+	conn, err := net.Dial("tcp", tcp.Address)
+	if err != nil {
+		a.logger.Println("E! failed to connect", err)
+		return
+	}
+
+	fmt.Fprintf(conn, bodyBuffer+"\n")
+
+	defer conn.Close()
```

Comment:
I actually copied this convention from the `handlePost` method [above](https://github.com/influxdata/kapacitor/pull/873/files/fe4e9d5bb92515f74b42a3034997332394e3fc56#diff-d02710009bb4a30f396667696f963627R915). @nathanielc might be able to clarify, but I assume this was added to ensure `Close()` is always called and to prevent the socket from remaining in ESTABLISHED for longer than it needs to.